### PR TITLE
test pickling

### PR DESCRIPTION
**Pull request by williamFalcon**
Pickling seems to be a problem more frequently. Tests added to test the API for pickling.

### CODE REVIEW

File: pytorch_lightning/callbacks/early_stopping.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def __init__(self, monitor: str = 'val_loss', min_delta: float = 0.0, patience:
         self.min_delta = min_delta
         self.wait = 0
         self.stopped_epoch = 0
+        self.mode = mode
 
         mode_dict = {
```
Comment (jeremyjordan) [inline pytorch_lightning/callbacks/early_stopping.py:62]: shouldn't this be deleted now that you defined it as a property?

**Comment by jeremyjordan** [comment]
the tests should also cover unpickling to verify that everything is reloaded properly

File: pytorch_lightning/callbacks/early_stopping.py (hunk 1 above)
Comment (jeremyjordan) [inline pytorch_lightning/callbacks/early_stopping.py:62]: don't like that this `mode_dict` is defined in two places now..

**Comment by Borda** [comment]
> the tests should also cover unpickling to verify that everything is reloaded properly

mind send followup PR? :rabbit:

**Pull request merged**